### PR TITLE
Added support for using as a clpm github dependency

### PR DESCRIPTION
**Pull request by ohmree**
This PR adds files that tell [clpm](clpm.dev) where to  find the non-quicklisp dependencies of this project, allowing it to be used as a github dependency in other projects' `clpmfile`s.

### CODE REVIEW

**Comment by digikar99** [comment]
Thanks a ton for the PR!

Not a CLPM user here; is the lock file essential if it is autogenerated? Or should it be gitignored? It seems to be tracking the dependencies - updates to the dependencies seems like it would require an update to the lock-file, and I'd want to avoid tracking the dependencies.

EDIT: Or if its essential for stability, is there a way to handle it on the user's side? 

At this point of the project, I think I'd rather have the project's main branch break due to slight-dependency-updates; OTOH, I'd be willing to provide stability either through ultralisp/quicklisp dists or a release.

**Comment by ohmree** [comment]
Well it's a lockfile - just like the one cargo or npm projects have.

So no, it's not necessary, and if you don't want to use clpm yourself and/or track dependency versions then you can just leave it out.

The most important thing for me is the clpmfile itself, since I got invited to work on a project that depends on polymorphic-functions and as a clpm user, since it doesn't support ultralisp, I'd like to be able to actually run the code I'll be working on.  

TL;DR: I'd love for just the clpmfile to be merged, it'll make my life a lot easier and serve as the basis for 7 other PRs adding clpm support to projects that depend on this.

**Comment by digikar99** [comment]
Thanks! Merged and removed the lockfile in 86280363172d598b1bd8173e43286cd493b3f0c2 and 636c47310ed865a0bb5fe73494be3885bb18b5d2

**Comment by ohmree** [comment]
Awesome!
Closing this.

**Comment by mfiano** [comment]
I'd like to chime in here. The lock file is actually pretty important to add. It is what ensures reliable reproducible builds. It is the lock file that is updated with newer loose dependency versions when you use CLPM's client API for updating a context. Without this lock file, everyone will pull newer versions.

Plus clpm doesn't descend into dependencies looking for clpmfiles. So if you're using this as a dependency, this file will do no good (other than as a reference to write a clpmfile in the top level project)
